### ISSUE TITLE
[Bug]: syntaxerror: Left side of assignment reference

### ISSUE THREAD

**Issue by sbhehe233** [stepci/stepci#202]
### What happened?

followed the bun sample (docs/integration/bun.md) and ran into the below syntax error

Tried both 1.12 and 2.0.0 and had the same outcome

![IMG_1319](https://github.com/stepci/stepci/assets/55295768/32e46aea-099b-42bf-89da-0260475019d7)

### What did you expect to happen?

Tests should run as expected

### Version

1.12

### Environment

Bun v1.026

### How can we reproduce this bug?

Run the official bun sample (docs/integration/bun.md) with Bun v1.026

### Would you be interested in working on a bugfix for this issue?

- [ ] Yes! Assign me

**Comment by mishushakov** [comment]
Hey, thanks for starting the issue!

It's a known issue with Bun, not related to Step CI.

**Comment by sbhehe233** [comment]
> Hey, thanks for starting the issue!
> 
> It's a known issue with Bun, not related to Step CI.

fair enough but shouldnt this be noted in the sample? ppl would think it’s supported while actually it isnt yet …

**Comment by mishushakov** [comment]
Yes, I think we should mention it in the docs. Can you contribute a fix to the doc?

**Comment by paperdave** [comment]
> It's a known issue with Bun

was there an open bug in bun before this one? sorry if it was missed. this seems to be a very trivial parser bug:

`(a?.b)[1] = 2;`

minimized from this line

`(stepResult.checks?.messages)[check.id] = {`

the optional chain is not handled by bun. i think we can fix this easily but i'm currently wrapped up in a lot of windows compat work.

**Comment by mishushakov** [comment]
Sorry, I meant that this is known issue among Step CI users using Bun, not Bun's issue in particular.
I'm not sure how the fix would look like.

Also, keep in mind that `got`, the library used by Step CI also does not work with Bun: https://github.com/oven-sh/bun/issues/3150

**Comment by paperdave** [comment]
ah. `got` would be the next blocker after this, not just the simple parsing bug.

i think it is reasonable to add in your docs that bun support is blocked by these issues (can open a new issue in bun specifically for stepci to link, and that issue should point to got+the parsing bug). it is our responsibility to fix these all.

**Comment by sbhehe233** [comment]
> Yes, I think we should mention it in the docs. Can you contribute a fix to the doc?

**Comment by mishushakov** [comment]
Yes!

**Comment by sbhehe233** [comment]
Now this has been taken care of with bun 1.0.27 but a new issue has occurred unfortunately… will follow up with another thread cheers
